vhdbuilder/publish/prepare-sig.sh

```diff
@@ -39,6 +39,10 @@ az account set --subscription ${SUBSCRIPTION_ID}
    --gallery-image-definition ${IMAGEDEFINITION_NAME})
  if [ -z "$id" ]; then
    echo "Creating image definition ${IMAGEDEFINITION_NAME} in Shared Image Gallery ${GALLERY_NAME} inside the resource group ${RG_NAME}"
+   HYPERV_GENERATION=V1
+   if [[ ${IMAGEDEFINITION_NAME} == *"gen2"* ]]; then
```

Comment:
This might be ok, but have we thought about taking a parameter? 
Maybe keep it an option in future.

Reply:
changed to use variable in pipeline